doc/API.rst.txt

```diff
@@ -125,6 +125,8 @@ change is made.
 
 * `scipy.optimize`
 
+  - `scipy.optimize.cython_optimize
```

Comment:
this seems to cause a CI failure: `API.rst.txt:128: WARNING: Inline interpreted text or phrase reference start-string without end-string.`

Reply:
That was fixed in https://github.com/scipy/scipy/pull/16806, but I don't know if anything other than the correction to `API.rst.txt` is OK to be backported.

Reply:
thanks, added the fix